src/Lean/Parser/Command.lean

```diff
@@ -18,6 +18,10 @@ namespace Parser
 @[builtinTermParser] def Term.quot := leading_parser "`(" >> incQuotDepth (termParser <|> many1Unbox commandParser) >> ")"
 @[builtinTermParser] def Term.precheckedQuot := leading_parser "`" >> Term.quot
 
+-- preserve dangling `.` in the syntax tree when failing
+-- NOTE: `ident >> checkNoWsBefore >> "."` only succeeds on dangling `.`, so full parser in `optional` can never succeed
+def identCompl := leading_parser ident >> optional (checkNoWsBefore >> "." >> checkNoWsBefore >> ident)
```

Comment:
@leodemoura I'd love it if there was a way for this to return an unwrapped `ident` on success, but I didn't find an appropriate encoding using the current combinators